### PR TITLE
Add logging about DB and connection pools

### LINKED ISSUE #2522: Add monitoring for DB activity
* DB pool client counts
 * function invocation counts (aggregate?)

### PR DESCRIPTION
**Pull request by djmitche**
The first of these two commits can get cherry-picked to `master` once #2570 lands.  The other adds support for passing monitor from each service, and will remain on the tc-lib-postgres branch until phase 1.

Github Bug/Issue: Fixes #2522

### CODE REVIEW

File: libraries/postgres/src/Database.js [hunk 1]
```diff
@@ -14,17 +15,49 @@ const EXTENSIONS = [
...
+    fields come from the node-postgres package.
+  `,
+  fields: {
+    pool: 'The name of the pool within the process',
+    totalCount: 'The total number of connections',
+    idleCount: 'The number of connections (out of the total) which are not currently in use',
+    waitingCount: 'The number of operations waiting for an idle connection',
+  },
+});
+
```
Comment (helfi92) [inline libraries/postgres/src/Database.js]: Should these be in a `monitor.js` file to be consistent with how the rest of services are written?

Comment (djmitche) [reply libraries/postgres/src/Database.js]: For tc-lib-pulse and tc-lib-api, they're in the file where the logging calls are made, so I just followed that pattern.

**Comment by djmitche** [comment]
@edunham, does this look useful to you?  In particular [this](https://github.com/taskcluster/taskcluster/pull/2571/files#diff-57f9f48e1fee6178bc651b965dd71599R18) is the docs for the two new log messages.

**Comment by djmitche** [comment]
(@helfi92, feel free to merge and cherry-pick this next week if it doesn't get merged today)

**Comment by helfi92** [comment]
cherry pick it to master?

**Comment by djmitche** [comment]
just the first commit -- the second commit is specific to the phase-1 work.

**Comment by helfi92** [comment]
Sounds good.

**Comment by helfi92** [comment]
Cherry picked the first commit in https://github.com/taskcluster/taskcluster/pull/2587.

**Comment by helfi92** [comment]
Rebase required.

**Comment by helfi92** [comment]
https://github.com/taskcluster/taskcluster/pull/2600

**Comment by edunham** [comment]
Looks good to me -- I assume it respects services' loglevel configs?

**Comment by djmitche** [comment]
> Looks good to me -- I assume it respects services' loglevel configs?

It does!

This has ended up merged from the tc-lib-postgres branch, so nothing more to do here.

**Pull request closed without merging**